eth/beacon/state_machines/forks/serenity/validation.py

```diff
@@ -21,12 +21,49 @@
 )
 from eth.beacon.helpers import (
     get_attestation_participants,
+    get_beacon_proposer_index,
     get_block_root,
     get_domain,
 )
+from eth.beacon.types.blocks import BaseBeaconBlock  # noqa: F401
 from eth.beacon.types.states import BeaconState  # noqa: F401
 from eth.beacon.types.attestations import Attestation  # noqa: F401
 from eth.beacon.types.attestation_data import AttestationData  # noqa: F401
+from eth.beacon.types.proposal_signed_data import ProposalSignedData
+from eth.beacon.typing import (
+    ShardNumber,
+)
+
+
+#
+# Proposer signature validation
+#
+def validate_serenity_proposer_signature(state: BeaconState,
+                                         block: BaseBeaconBlock,
+                                         beacon_chain_shard_number: ShardNumber,
+                                         epoch_length: int) -> None:
+    block_without_signature_root = block.block_without_signature_root
+
+    # TODO: Replace this root with tree hash root
+    proposal_root = ProposalSignedData(
+        state.slot,
+        beacon_chain_shard_number,
+        block_without_signature_root,
+    ).root
+
+    # Get the public key of proposer
+    beacon_proposer_index = get_beacon_proposer_index(state, state.slot, epoch_length)
+    proposer_pubkey = state.validator_registry[beacon_proposer_index].pubkey
```

Comment:
I'm always a little nervous with indexing into iterables naively like this.  I understand/recognize that the previous `get_beacon_proposer_index` provides some level of guarantee that the `beacon_proposer_index` is indeed present and accurate in the `validator_registry` iterable.

What if this was abstracted away on the `BeaconState` class as something like `state.get_beacon_proposer(epoch_length)`:

```python
proposer = state.get_beacon_proposer(epoch_length)
proposer_pubkey = proposer.pubkey
```